src/main/java/org/folio/search/service/converter/SearchDocumentConverter.java

```diff
@@ -114,18 +116,33 @@ private static Object getFieldValue(
     return getObjectFieldValue(objectMapValue, objectFieldDescription.getProperties(), ctx);
   }
 
-  private static Object getPlainFieldValue(Map<String, Object> fieldData,
+  private Object getPlainFieldValue(Map<String, Object> fieldData,
     Entry<String, FieldDescription> fieldEntry, ConversionContext ctx) {
     var desc = (PlainFieldDescription) fieldEntry.getValue();
     if (isNotIndexedField(desc)) {
       return null;
     }
-    Object plainFieldValue = fieldData.get(fieldEntry.getKey());
+    Object plainFieldValue = getPlainFieldValue(fieldData, fieldEntry.getKey(), desc);
     return isMultilangField(desc) ? getMultilangValue(plainFieldValue, ctx) : plainFieldValue;
   }
 
+  private Object getPlainFieldValue(Map<String, Object> fieldData, String fieldName,
+    PlainFieldDescription desc) {
+
+    if (!desc.hasPopulatedBy()) {
+      return fieldData.get(fieldName);
+    }
+
+    final var propertySetter = fieldSetters.get(desc.getPopulatedBy());
+    if (propertySetter == null) {
+      throw new IllegalArgumentException("There is no such property setter: " + desc.getPopulatedBy());
```

Comment:
throwing exception in terms of this type of indexing will fail the WHOLE batch, consider to refactor it

Reply:
Thanks, good catch. Do you think it is fine just a log warn message?

Reply:
I guess that this check can be removed, it looks like as metadata validation. From my viewpoint, this should be done at the application startup or resource description loaded event. If metadata is not valid - user or developer should receive warning single time, not for each event in kafka.

Reply:
Thanks, will do so.

Reply:
Changed, the validation is part of `ResourceDescriptionService#init` logic now.